regolith/resolver.go

```diff
@@ -52,123 +51,155 @@ func resolveResolverUrl(url string) (string, error) {
 	return fmt.Sprintf("git::https://%s/%s", repoUrl, path), nil
 }
 
-// DownloadResolverMaps downloads the resolver.json files
-func DownloadResolverMaps() error {
+// resolveResolverUrl resolves the resolver URL from the short name to a full
+// URL that can be used by go-getter
+func resolveResolverUrlNew(url string) (string, string, error) {
+	urlParts := strings.Split(url, "/")
+	if len(urlParts) < 4 {
+		return "", "", burrito.WrappedErrorf(
+			"Incorrect URL format.\n" +
+				"Expected format:" +
+				"github.com/<user-name>/<repo-name>/<path-to-the-resolver-file>")
+	}
+	repoUrl := strings.Join(urlParts[0:3], "/")
+	path := strings.Join(urlParts[3:], "/")
+	return fmt.Sprintf("https://%s", repoUrl), path, nil
+}
+
+// DownloadResolverMaps downloads the resolver repositories and returns lists of urls and paths
+func DownloadResolverMaps(forceUpdate bool) ([]string, []string, error) {
 	Logger.Info("Downloading resolvers")
 
 	// Define function to download group of resolvers
-	downloadResolvers := func(urls []string, root string) error {
-		MeasureStart("Prepare for resolvers download")
-		targetPath := filepath.Join(root, "resolvers")
-		tmpPath := filepath.Join(root, ".resolvers-tmp")
-		tmpResolversPath := filepath.Join(tmpPath, "resolvers")
-		tmpUndoPath := filepath.Join(tmpPath, "undo")
-		// Create target directory if not exists
-		err := os.MkdirAll(targetPath, 0755)
-		if err != nil {
-			return burrito.WrapErrorf(err, osMkdirError, targetPath)
+	downloadResolvers := func(urls []string, root string) ([]string, error) {
```

Comment:
I don't remember why this was implemented as a closure. It was probably reused multiple times in the function in the original implementation but now it's just used once. The `develop` branch also uses it once. I didn't read into it too much so maybe I'm missing something.

Reply:
Made it part of the code instead of a closure.